Micronaut AOT does not work with Test Resources

### Expected Behavior

When I generate a vanilla Micronaut project with Test Resources, like this:

```
mn create-app example.micronaut.test-resources-fail --features=test-resources
```

... it should be able to build.

### Actual Behaviour

The build fails, see log below.

If I remove the `aot` section from the `micronaut` section in build.gradle, the build still fails. If I also remove the `io.micronaut.aot` plugin from the `plugins` section, the build works.

Note: I moved this issue from the Micronaut Test Resources project, as it seems to be a bug in Micronaut AOT and not Test Resources. Original: https://github.com/micronaut-projects/micronaut-test-resources/issues/386

```
% ./gradlew build      
[test-resources-service] 13:10:54.597 [ForkJoinPool.commonPool-worker-7] INFO  i.m.t.e.TestResourcesResolverLoader - Loaded 3 test resources resolvers: io.micronaut.testresources.redis.RedisTestResourceProvider, io.micronaut.testresources.redis.RedisClusterTestResourceProvider, io.micronaut.testresources.testcontainers.GenericTestContainerProvider

> Task :internalStartTestResourcesService
[1.113s][warning][cds] Preload Warning: Cannot find io/netty/channel/$Proxy12
[1.339s][warning][cds] Skipping jdk/internal/event/Event: JFR event class
[1.341s][warning][cds] Skipping jdk/internal/event/SecurityProviderServiceEvent: JFR event class
[1.717s][warning][gc,verify] (Unmovable) humongous regions have been found and may lead to fragmentation while writing archive heap memory regions.

[test-resources-service] 13:10:54.688 [main] INFO  i.m.t.server.TestResourcesService - A Micronaut Test Resources server is listening on port 52730, started in 139ms

> Task :prepareJitOptimizations FAILED
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
        at io.micronaut.aot.cli.Main.executeInIsolatedLoader(Main.java:113)
        at io.micronaut.aot.cli.Main.run(Main.java:87)
        at picocli.CommandLine.executeUserObject(CommandLine.java:2026)
        at picocli.CommandLine.access$1500(CommandLine.java:148)
        at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2415)
        at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273)
        at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)
        at picocli.CommandLine.execute(CommandLine.java:2170)
        at io.micronaut.aot.cli.Main.execute(Main.java:134)
        at io.micronaut.aot.cli.Main.main(Main.java:138)
Caused by: java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at io.micronaut.aot.cli.Main.executeInIsolatedLoader(Main.java:107)
        ... 11 more
Caused by: io.micronaut.testresources.client.TestResourcesException: java.net.ConnectException
        at io.micronaut.testresources.client.DefaultTestResourcesClient.request(DefaultTestResourcesClient.java:144)
        at io.micronaut.testresources.client.DefaultTestResourcesClient.getRequiredPropertyEntries(DefaultTestResourcesClient.java:99)
        at java.base/java.util.Optional.map(Optional.java:260)
        at io.micronaut.testresources.client.TestResourcesClientPropertySourceLoader$ClientTestResourcesResolver.getPropertyEntries(TestResourcesClientPropertySourceLoader.java:50)
        at io.micronaut.testresources.core.LazyTestResourcesPropertySourceLoader$LazyPropertySource.computeKeys(LazyTestResourcesPropertySourceLoader.java:98)
        at io.micronaut.testresources.core.LazyTestResourcesPropertySourceLoader$LazyPropertySource.iterator(LazyTestResourcesPropertySourceLoader.java:90)
        at io.micronaut.context.env.PropertySourcePropertyResolver.processPropertySource(PropertySourcePropertyResolver.java:596)
        at io.micronaut.context.env.DefaultEnvironment.readPropertySources(DefaultEnvironment.java:449)
        at io.micronaut.context.env.DefaultEnvironment.start(DefaultEnvironment.java:278)
        at io.micronaut.context.DefaultApplicationContext$RuntimeConfiguredEnvironment.start(DefaultApplicationContext.java:881)
        at io.micronaut.context.DefaultApplicationContext$RuntimeConfiguredEnvironment.start(DefaultApplicationContext.java:850)
        at io.micronaut.aot.core.context.ApplicationContextAnalyzer.finalizeConfiguration(ApplicationContextAnalyzer.java:114)
        at io.micronaut.aot.core.context.ApplicationContextAnalyzer.create(ApplicationContextAnalyzer.java:91)
        at io.micronaut.aot.MicronautAotOptimizer$Runner.execute(MicronautAotOptimizer.java:328)
        at io.micronaut.aot.MicronautAotOptimizer.execute(MicronautAotOptimizer.java:202)
        ... 16 more
Caused by: java.net.ConnectException
        at java.net.http/jdk.internal.net.http.HttpClientImpl.send(HttpClientImpl.java:573)
        at java.net.http/jdk.internal.net.http.HttpClientFacade.send(HttpClientFacade.java:123)
        at io.micronaut.testresources.client.DefaultTestResourcesClient.request(DefaultTestResourcesClient.java:134)
        ... 30 more
Caused by: java.net.ConnectException
        at java.net.http/jdk.internal.net.http.common.Utils.toConnectException(Utils.java:1055)
        at java.net.http/jdk.internal.net.http.PlainHttpConnection.connectAsync(PlainHttpConnection.java:198)
        at java.net.http/jdk.internal.net.http.PlainHttpConnection.checkRetryConnect(PlainHttpConnection.java:230)
        at java.net.http/jdk.internal.net.http.PlainHttpConnection.lambda$connectAsync$1(PlainHttpConnection.java:206)
        at java.base/java.util.concurrent.CompletableFuture.uniHandle(CompletableFuture.java:934)
        at java.base/java.util.concurrent.CompletableFuture$UniHandle.tryFire(CompletableFuture.java:911)
        at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510)
        at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1773)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.nio.channels.ClosedChannelException
        at java.base/sun.nio.ch.SocketChannelImpl.ensureOpen(SocketChannelImpl.java:195)
        at java.base/sun.nio.ch.SocketChannelImpl.beginConnect(SocketChannelImpl.java:760)
        at java.base/sun.nio.ch.SocketChannelImpl.connect(SocketChannelImpl.java:848)
        at java.net.http/jdk.internal.net.http.PlainHttpConnection.lambda$connectAsync$0(PlainHttpConnection.java:183)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:569)
        at java.net.http/jdk.internal.net.http.PlainHttpConnection.connectAsync(PlainHttpConnection.java:185)
        ... 9 more

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':prepareJitOptimizations'.
> Process 'command '/Library/Java/JavaVirtualMachines/temurin-17.jdk/Contents/Home/bin/java'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/8.2.1/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 3s
11 actionable tasks: 2 executed, 9 up-to-date
```

### Steps To Reproduce

1. Create a Micronaut application with `mn create-app example.micronaut.test-resources-fail --features=test-resources`
2. Run `./gradlew build`

### Environment Information

OS: MacOS Ventura 13.5.2
JDK: OpenJDK Temurin 17.0.8

### Example Application

https://github.com/runenielsen/test-resources-fail

### Version

4.1.0

--- a/aot-plugin/src/main/java/io/micronaut/gradle/aot/MicronautAotPlugin.java
+++ b/aot-plugin/src/main/java/io/micronaut/gradle/aot/MicronautAotPlugin.java
@@ -50,7 +50,6 @@ import org.gradle.api.plugins.JavaApplication;
 import org.gradle.api.plugins.JavaPlugin;
 import org.gradle.api.provider.Provider;
 import org.gradle.api.provider.ProviderFactory;
-import org.gradle.api.specs.Spec;
 import org.gradle.api.tasks.JavaExec;
 import org.gradle.api.tasks.SourceSet;
 import org.gradle.api.tasks.TaskContainer;
@@ -447,48 +446,6 @@ public abstract class MicronautAotPlugin implements Plugin<Project> {
         private static String normalizePath(String path) {
             return path.replace('\\', '/');
         }
-    }
-
-    private static class ExclusionSpec implements Spec<File> {
-        private final Provider<RegularFile> filterFile;
-        private final Set<String> prefixes;
-        private final Logger logger;
-        private Set<String> excludes;
-
-        private ExclusionSpec(Provider<RegularFile> filterFile,
-                                 Set<String> prefixes,
-                                 Logger logger) {
-            this.filterFile = filterFile;
-            this.prefixes = prefixes;
-            this.logger = logger;
-        }
-
-        @Override
-        public boolean isSatisfiedBy(File file) {
-            if (excludes == null) {
-                File resourceFilter = filterFile.get().getAsFile();
-                try {
-                    excludes = new HashSet<>();
-                    Files.readAllLines(resourceFilter.toPath())
-                            .stream()
-                            .map(JarExclusionSpec::normalizePath)
-                            .forEach(excludes::add);
-                } catch (IOException e) {
-                    throw new RuntimeException(e);
-                }
-                logger.debug("Excluded resources: {} ", excludes);
-            }
-            var relativePath = file.toString();
-            if (excludes.contains(normalizePath(relativePath)) || prefixes.stream().anyMatch(p -> normalizePath(relativePath).startsWith(p))) {
-                return false;
-            }
-            return true;
-        }
-
-        private static String normalizePath(String path) {
-            return path.replace('\\', '/');
-        }
-
     }
 
     private static final class Configurations {

--- a/functional-tests/src/test/groovy/io/micronaut/gradle/testresources/TestResourcesWithAotSpec.groovy
+++ b/functional-tests/src/test/groovy/io/micronaut/gradle/testresources/TestResourcesWithAotSpec.groovy
@@ -50,4 +50,35 @@ micronaut {
         !result.output.contains("Configuration with name 'optimizedRuntimeClasspath' not found.")
         result.task(':help').outcome == TaskOutcome.SUCCESS
     }
+
+    @Issue("https://github.com/micronaut-projects/micronaut-gradle-plugin/issues/852")
+    def "test resources client shouldn't be on the AOT classpath"() {
+        settingsFile << "rootProject.name = 'test'"
+        buildFile << """plugins {
+    id("io.micronaut.minimal.library")
+    id("io.micronaut.aot")
+    id("io.micronaut.test-resources")
+}
+
+repositories {
+    mavenCentral()
+}
+
+micronaut {
+    version "$micronautVersion"
+}
+
+tasks.named("prepareJitOptimizations") {
+    doFirst {
+        assert classpath.files.stream().noneMatch { it.name.startsWith("micronaut-test-resources-client-") }
+    }
+}
+
+"""
+        when:
+        def result = build 'prepareJitOptimizations'
+
+        then:
+        result.task(":prepareJitOptimizations").outcome == TaskOutcome.SUCCESS
+    }
 }

--- a/test-resources-plugin/src/main/java/io/micronaut/gradle/testresources/internal/TestResourcesAOT.java
+++ b/test-resources-plugin/src/main/java/io/micronaut/gradle/testresources/internal/TestResourcesAOT.java
@@ -36,8 +36,6 @@ public final class TestResourcesAOT {
     public static void configure(Project project, Configuration client) {
         AOTExtension aot = PluginsHelper.findMicronautExtension(project).getExtensions().getByType(AOTExtension.class);
         ConfigurationContainer configurations = project.getConfigurations();
-        Configuration aotAppClasspath = configurations.getByName(MicronautAotPlugin.AOT_APPLICATION_CLASSPATH);
-        aotAppClasspath.extendsFrom(client);
         project.getPluginManager().withPlugin("io.micronaut.minimal.application", unused -> {
             Configuration optimizedRuntimeClasspath = configurations.getByName(MicronautAotPlugin.OPTIMIZED_RUNTIME_CLASSPATH_CONFIGURATION_NAME);
             optimizedRuntimeClasspath.extendsFrom(client);

